packages/composite-checkout/src/components/billing-fields.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useLocalize } from '../lib/localize';
+import { useCheckoutLineItems } from '../index';
+import GridRow from './grid-row';
+import Field from './field';
+
+export default function BillingFields( {
+	summary,
+	setPaymentData,
+	paymentData,
+	isActive,
+	isComplete,
+} ) {
+	const [ items ] = useCheckoutLineItems();
+	const localize = useLocalize();
+	if ( summary && isComplete ) {
+		return <BillingFormSummary paymentData={ paymentData } />;
+	}
+	if ( ! isActive ) {
+		return null;
+	}
+	const {
+		billingName = '',
+		billingAddress = '',
+		billingCity = '',
+		billingState = '',
+		billingZipCode = '',
+		billingCountry = '',
+		billingPhoneNumber = '',
+		billingVatIdNumber = '',
+		billingNameError = null,
+		billingAddressError = null,
+		billingCityError = null,
+		billingStateError = null,
+		billingZipCodeError = null,
+		billingCountryError = null,
+		billingPhoneNumberError = null,
+		billingVatIdError = null,
+	} = paymentData || {};
+
+	const nameOnChange = value => setPaymentData( { ...( paymentData || {} ), billingName: value } );
+	const addressOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingAddress: value } );
+	const cityOnChange = value => setPaymentData( { ...( paymentData || {} ), billingCity: value } );
+	const stateOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingState: value } );
+	const zipCodeOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingZipCode: value } );
+	const countryOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingCountry: value } );
+	const phoneNumberOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingPhoneNumber: value } );
+	const vatIdOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingVatIdNumber: value } );
+
+	return (
+		<BillingFormFields>
+			{ isDomainRelatedField( items ) && (
```

Comment:
I would suggest putting this fragment into its own component, so then this line would read

```
{ isDomainRelatedField( items ) && <DomainFields setPaymentData={ setPaymentData } paymentData={ paymentData } /> }
```

Reply:
Good idea. Thanks for the tip.